nodebuilder/daser/config.go

```diff
@@ -0,0 +1,44 @@
+package daser
+
+import (
+	"errors"
+	"time"
+)
+
+var (
+	ErrNegativeInterval = errors.New("nodebuilder/daser: interval must be positive")
+)
+
+type Config struct {
+	//  samplingRange is the maximum amount of headers processed in one job.
+	SamplingRange uint
+
+	// concurrencyLimit defines the maximum amount of sampling workers running in parallel.
+	ConcurrencyLimit uint
+
+	// backgroundStoreInterval is the period of time for background checkpointStore to perform a checkpoint backup.
+	BackgroundStoreInterval time.Duration
+
+	// priorityQueueSize defines the size limit of the priority queue
+	PriorityQueueSize uint
+
+	// genesisHeight is the height sampling will start from
+	GenesisHeight uint
+}
+
+// TODO(@derrandz): parameters needs performance testing on real network to define optimal values
+func DefaultConfig() Config {
+	return Config{
+		SamplingRange:           100,
+		ConcurrencyLimit:        16,
+		BackgroundStoreInterval: 10 * time.Minute,
+		PriorityQueueSize:       16 * 4,
+		GenesisHeight:           1,
+	}
+}
+
+// Validate performs basic validation of the config.
+func (cfg *Config) Validate() error {
+	// TODO(team): what should validate for in here? seems like all configuration fields can accept zero-values.
```

Comment:
you can ensure all fields are > 0

Reply:
they are `unit` so the conditional will always be truthy even if assigned `-1`

Reply:
Either of `SamplingRange` == 0 (empty jobs) OR `ConcurrencyLimit` == 0 (no workers) OR `GenesisHeight ` == 0 (can't sample block height 0) will break DASer.
`BackgroundStoreInterval` == 0 disables background storer,
`PriorityQueueSize` == 0 disables prioritisation of recently produced blocks for sampling 

Lets protect user from breaking the DASer by passing 0 to config fields that could not support it.